src/go/rpk/pkg/cli/debug/remotebundle/download.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package remotebundle
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/debug/common"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+)
+
+type downloadResponse struct {
+	Broker     string
+	Downloaded bool
+	Error      string
+}
+
+func newDownloadCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		noConfirm bool
+		jobID     string
+		outFile   string
+	)
+	cmd := &cobra.Command{
+		Use:   "download",
+		Short: "Download a remote debug bundle",
+		Long: `Download a remote debug bundle.
+
+This command will download the debug collection process in a remote cluster that
+you have configured either in your rpk profile, flags or environment variables.
+
+Alternatively, you may provide a --job-id to only download the debug bundle with
+the given Job ID.
+
+The flag '--no-confirm' can be used to avoid the confirmation prompt.
```

Comment:
```suggestion
This command downloads the debug collection process in a remote cluster that
you configured in flags, environment variables, or your rpk profile.

Use the flag '--job-id' to only download the debug bundle with
the given job ID.

Use the flag '--no-confirm' to avoid the confirmation prompt.
```